qrand/platforms/qsharp/job.py

```diff
@@ -31,34 +34,93 @@
+    ) -> None:
+        self.backend = backend
+        self.circuit = circuit
+        self.num_measurements: int = num_measurements  # type: ignore
 
     ############################### PUBLIC API ###############################
     @property
     def backend(self) -> QsharpBackend:
-        raise NotImplementedError(self.ERROR_MSG)
+        return self._backend
 
     @backend.setter
     def backend(self, backend: QsharpBackend) -> None:
-        raise NotImplementedError(self.ERROR_MSG)
+        self._backend: QsharpBackend = backend
 
     @property
     def circuit(self) -> QsharpCircuit:
-        raise NotImplementedError(self.ERROR_MSG)
+        return self._circuit
 
     @circuit.setter
     def circuit(self, circuit: QsharpCircuit) -> None:
-        raise NotImplementedError(self.ERROR_MSG)
+        if self.backend.max_qubits < circuit.num_qubits:
+            raise RuntimeError(
+                f"Failed to assign QsharpCircuit for QsharpJob. Number of \
+                qubits in QsharpCircuit unsupported by this job's Backend: \
+                {self.backend.max_qubits}<{circuit.num_qubits}."
+            )
+        self._circuit: QsharpCircuit = circuit
 
     @property
-    def num_measurements(self) -> int:
-        raise NotImplementedError(self.ERROR_MSG)
+    def num_measurements(self) -> int:  # type: ignore
+        return self._num_measurements
 
     @num_measurements.setter
     def num_measurements(self, num_measurements: Optional[int]) -> None:
-        raise NotImplementedError(self.ERROR_MSG)
+        num_measurements = (
+            num_measurements
+            if isinstance(num_measurements, int) and 0 < num_measurements
+            else self.backend.max_measurements
+        )
+        if self.backend.max_measurements < num_measurements:
+            warn(
+                f"Number of measurements unsupported by the job's Backend: \
+                {self.backend.max_measurements}<{num_measurements}. \
+                Using max_measurements instead.",
+                UserWarning,
+            )
+            self._num_measurements = self.backend.max_measurements
+
+    def _generate_code(self) -> QSharpCallable:
+        qsharp_code = """
+        open Microsoft.Quantum.Intrinsic;
+        open Microsoft.Quantum.Measurement;
+        open Microsoft.Quantum.Arrays;
+        open Microsoft.Quantum.Math;
+
+        operation Program():String[]{{
+
+        use q = Qubit[{num_qubits}];
+        mutable value = ConstantArray({num_qubits},"0");
+
+        {gates}
+
+        ResetAll(q);
+        return value;
+        }}
+        """
+        return compile(
+            qsharp_code.format(
+                num_qubits=self.circuit.num_qubits, gates=self.circuit.gates
+            )
+        )
 
     def execute(self) -> List[str]:
-        raise NotImplementedError(self.ERROR_MSG)
+        self.program = self._generate_code()
+
+        if self.backend.resource_id is None or self.backend.target_id is None:
```

Comment:
Is the `resource_id` set anywhere for this?

Reply:
No, it will be input by the user and will be initialized in QsharpJob.

Reply:
It is input for the creation of `QsharpBackend` and read from there in `QsharpJob`. Are we missing something @crazy4pi314?